addons/mail/wizard/mail_compose_message.py

```diff
@@ -283,7 +283,7 @@ def send_mail(self, auto_commit=False):
                         else:
                             ActiveModel.browse(res_id).message_post(**post_params)
 
-                if wizard.composition_mode == 'mass_mail':
+                if wizard.composition_mode == 'comment':  # Send e-mails immediately only when we are handling a single document (sent from the mail form)
                     batch_mails.send(auto_commit=auto_commit)
```

Comment:
This is more likely the fix you're looking for:

```suggestion
                if wizard.composition_mode == 'mass_mail':
                    batch_mails.mark_outgoing()
```

This way you mark the mails to be sent on next queue flush instead of doing nothing with them.

I hope this is acceptable for a stable version... otherwise we'd maybe need a context key to handle these cases specifically:

```suggestion
                if wizard.composition_mode == 'mass_mail':
                    if self.env.context.get('queue_mass_mail'):
                        batch_mails.mark_outgoing()
                    else:
                        batch_mails.send(auto_commit=auto_commit)
```

Odoo guys will say. Ping @tde-banana-odoo

Reply:
Thanks, that seems more appropriate, indeed. I made the first suggested change, but I can make further changes if needed.